File: MiniDOMBase/DMD/Anagrafica/Uffici/CUtentiXUfficioCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica



    ''' <summary>
    ''' Cursore sulla tabella degli uffici
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable>
    Public Class CUtentiXUfficioCursor
        Inherits DBObjectCursorBase(Of CUtenteXUfficio)

        Private m_IDUtente As CCursorField(Of Integer)
        Private m_IDUfficio As CCursorField(Of Integer)

        Public Sub New()
            Me.m_IDUtente = New CCursorField(Of Integer)("Utente")
            Me.m_IDUfficio = New CCursorField(Of Integer)("Ufficio")
        End Sub

        Public ReadOnly Property IDUtente As CCursorField(Of Integer)
            Get
                Return Me.m_IDUtente
            End Get
        End Property

        Public ReadOnly Property IDUfficio As CCursorField(Of Integer)
            Get
                Return Me.m_IDUfficio
            End Get
        End Property

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New CUtenteXUfficio
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_UtentiXUfficio"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return APPConn
        End Function
    End Class


End Class

File: MiniDOMBase/DMD/Anagrafica/Uffici/UtentiXUfficio/CUtentiXUfficioCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;

namespace minidom
{
    public partial class Anagrafica
    {



        /// <summary>
        /// Cursore di oggetti di tipo <see cref="CUtenteXUfficio"/>
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CUtentiXUfficioCursor 
            : minidom.Databases.DBObjectCursorBase<CUtenteXUfficio>
        {
            private DBCursorField<int> m_IDUtente;
            private DBCursorField<int> m_IDUfficio;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CUtentiXUfficioCursor()
            {
                m_IDUtente = new DBCursorField<int>("Utente");
                m_IDUfficio = new DBCursorField<int>("Ufficio");
            }

            /// <summary>
            /// ID utente
            /// </summary>
            public DBCursorField<int> IDUtente
            {
                get
                {
                    return m_IDUtente;
                }
            }

            /// <summary>
            /// ID ufficio
            /// </summary>
            public DBCursorField<int> IDUfficio
            {
                get
                {
                    return m_IDUfficio;
                }
            }

            /// <summary>
            /// Istanzia un nuovo oggetto
            /// </summary>
            /// <param name="dbRis"></param>
            /// <returns></returns>
            public override CUtenteXUfficio InstantiateNewT(DBReader dbRis)
            {
                return new CUtenteXUfficio();
            }

            //public override string GetTableName()
            //{
            //    return "tbl_UtentiXUfficio";
            //}

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.Uffici.UfficiConsentiti;
            }

            //protected internal override Databases.CDBConnection GetConnection()
            //{
            //    return Databases.APPConn;
            //}
        }
    }
}